src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/x4decoder/HaivisionX4DecoderCommunicator.java

```diff
@@ -0,0 +1,755 @@
+		stats.put(streamStatisticGroup + StreamMonitoringMetric.DROPPED_PACKETS.getName(), checkForNullData(streamStats.getDroppedPackets()));
+		stats.put(streamStatisticGroup + StreamMonitoringMetric.STREAM_LATENCY.getName(), checkForNullData(streamInfo.getLatency()));
+
+		if (srt != null) {
+			stats.put(streamStatisticGroup + StreamMonitoringMetric.ENCRYPTION.getName(), checkForNullData(srt.getEncryption()));
+			stats.put(streamStatisticGroup + StreamMonitoringMetric.KEY_LENGTH.getName(), checkForNullData(srt.getKeyLength()));
+			stats.put(streamStatisticGroup + StreamMonitoringMetric.DECRYPT_STATE.getName(), checkForNullData(srt.getDecryptState()));
+			stats.put(streamStatisticGroup + StreamMonitoringMetric.LOST_PACKETS.getName(), checkForNullData(srt.getLostPackets()));
+			stats.put(streamStatisticGroup + StreamMonitoringMetric.SENT_ACKS.getName(), checkForNullData(srt.getSentAcks()));
+			stats.put(streamStatisticGroup + StreamMonitoringMetric.SENT_NAKS.getName(), checkForNullData(srt.getSentNaks()));
+			stats.put(streamStatisticGroup + StreamMonitoringMetric.PATH_MAX_BANDWIDTH.getName(), checkForNullData(srt.getPathMaxBandwidth()));
+			stats.put(streamStatisticGroup + StreamMonitoringMetric.RTT.getName(), checkForNullData(srt.getRtt()));
+			stats.put(streamStatisticGroup + StreamMonitoringMetric.BUFFER.getName(), checkForNullData(srt.getBuffer()));
+			stats.put(streamStatisticGroup + StreamMonitoringMetric.SRT_LATENCY.getName(), checkForNullData(srt.getLatency()));
+		}
+	}
+
+	/**
+	 * This method is used update stream statistic
+	 *
+	 * @param stats list statistics property
+	 * @param adapterProperties list adapterProperties element
+	 * @param filteringErrorGroup filtering error group
+	 * @param adapterPropertiesName Name of adapterProperties
+	 */
+	private void updateInvalidAdapterProperties(Map<String, String> stats, Set<String> adapterProperties, String filteringErrorGroup, String adapterPropertiesName) {
+		Objects.requireNonNull(adapterProperties);
+		String streamStatisticGroup = filteringErrorGroup + DecoderConstant.HASH;
+
+		for (String adapterProperty : adapterProperties) {
+			stats.put(streamStatisticGroup + adapterPropertiesName + DecoderConstant.SPACE + adapterProperty, DecoderConstant.INVALID_INPUT);
+		}
+	}
+
+	/**
+	 * This method is used to retrieve streams statistics by send GET request to http://{IP_Address}apis/streams
+	 *
+	 * @param stats list statistics property
+	 *
+	 * When sessionID is null, the failedMonitor is going to update
+	 * When there is no response data, the failedMonitor is going to update
+	 * When there is an exception, the failedMonitor is going to update and exception is not populated
+	 */
+	private void retrieveStreamStats(Map<String, String> stats) {
+		Set<String> setStreamName = new HashSet<>();
+		Set<String> setStreamStatus = new HashSet<>();
+		Set<String> setPortNumber = new HashSet<>();
+
+		// Retrieve Adapter Properties
+		if (this.streamName != null) {
```

Comment:
you will have the setStreamName etc. after the first interval, please store it to reuse instead of handled it every interval. It's not necessary. Please also careful because I see you remove the value from the set. It should have a global variable, and local variable as well

Reply:
I fixed it